src/test/kotlin/net/atos/zac/app/ontkoppeldedocumenten/OntkoppeldeDocumentenRESTServiceTest.kt

```diff
@@ -0,0 +1,202 @@
+
+            Then("the exception from OpenZaak is rethrown") {
+                exception.fout.status shouldBe 400
+            }
+        }
+    }
+
+    Given("a document that exists in the database but results in a informatieobject with a zaak id from OpenZaak") {
+        val werklijstRechten = createWerklijstRechten(ontkoppeldeDocumentenVerwijderen = true)
+        val document = OntkoppeldDocument()
+        document.documentUUID = UUID.randomUUID()
+        document.id = 1
+        val informatieObject = EnkelvoudigInformatieObject()
+        val zaakInformatieObject = ZaakInformatieobject()
+        zaakInformatieObject.zaak = URI.create("https://example.com/${UUID.randomUUID()}")
+        every {
+            policyService.readWerklijstRechten()
+        } returns werklijstRechten
+        every {
+            ontkoppeldeDocumentenService.find(document.id)
+        } returns Optional.of(document)
+        every {
+            drcClientService.readEnkelvoudigInformatieobject(document.documentUUID)
+        } returns informatieObject
+        every {
+            zrcClientService.listZaakinformatieobjecten(informatieObject)
+        } returns mutableListOf(zaakInformatieObject)
+
+        When("the delete endpoint is called with the id of that document") {
+            val exception = shouldThrow<IllegalStateException> { sut.delete(document.id) }
+
+            Then("the an IllegalStateException should be thrown") {
+                exception shouldNotBe null
+            }
+        }
+    }
+
+    Given("a document that exists in the database and results in a informatieobject without a zaak id from OpenZaak") {
+        val werklijstRechten = createWerklijstRechten(ontkoppeldeDocumentenVerwijderen = true)
+        val document = OntkoppeldDocument()
+        document.documentUUID = UUID.randomUUID()
+        document.id = 1
+        val informatieObject = EnkelvoudigInformatieObject()
+        val zaakInformatieObject = ZaakInformatieobject()
+        zaakInformatieObject.zaak = URI.create("https://example.com/${UUID.randomUUID()}")
+        every {
+            policyService.readWerklijstRechten()
+        } returns werklijstRechten
+        every {
+            ontkoppeldeDocumentenService.find(document.id)
+        } returns Optional.of(document)
+        every {
+            drcClientService.readEnkelvoudigInformatieobject(document.documentUUID)
+        } returns informatieObject
+        every {
+            drcClientService.deleteEnkelvoudigInformatieobject(document.documentUUID)
+        } just runs
+        every {
+            zrcClientService.listZaakinformatieobjecten(informatieObject)
+        } returns mutableListOf()
+        every {
+            ontkoppeldeDocumentenService.delete(document.id)
+        } just runs
+
+        When("the delete endpoint is called with the id of that document") {
+            sut.delete(document.id)
+
+            Then("the document is deleted") {
+                verify {
```

Comment:
Verify number of calls?
```suggestion
                verify(exactly = 1) {
```